### PR TITLE
Add new nuget resolver

### PR DESCRIPTION
**Pull request by JonDouglas**
adds new nuget resolver to what's new in SDK.

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/core/whats-new/dotnet-9/sdk.md](https://github.com/dotnet/docs/blob/6d849a8b75861b5ad991fd964f8b9d603df15830/docs/core/whats-new/dotnet-9/sdk.md) | [docs/core/whats-new/dotnet-9/sdk](https://review.learn.microsoft.com/en-us/dotnet/core/whats-new/dotnet-9/sdk?branch=pr-en-us-43302) |

### CODE REVIEW

File: docs/core/whats-new/dotnet-9/sdk.md [hunk 1]
```diff
@@ -115,6 +115,12 @@ If you have feedback about the terminal logger, you can provide it in the [MSBui
 
 Starting in .NET 8, `dotnet restore` [audits NuGet package references for known vulnerabilities](../../tools/dotnet-restore.md#audit-for-security-vulnerabilities). In .NET 9, the default mode has changed from auditing only _direct_ package references to auditing both _direct_ and _transitive_ package references.
 
+## Faster NuGet dependency resolution for large repos
+
+The NuGet dependency resolver has been overhauled to improve performance and scalability for all `<PackageReference>` projects. Enabled by default, the new algorithm speeds up restore operations without compromising on functionality, strictly adhering to the core dependency resolution rules.
+
+If you encounter any issues, such as restore failures or unexpected package versions compared to previous releases, you can [revert to the legacy resolver](https://learn.microsoft.com/nuget/consume-packages/Package-References-in-Project-Files#nuget-dependency-resolver).
```
Comment (nkolev92) [inline docs/core/whats-new/dotnet-9/sdk.md:122]: This is great. We will have a known issues section in 6.12 release notes as well, do you think it's something we should link from here?

Comment (JonDouglas) [reply docs/core/whats-new/dotnet-9/sdk.md:122]: maybe we update the central docs that this links to w/ that known issues link?

**Review by nkolev92 — APPROVED** [review]

File: docs/core/whats-new/dotnet-9/sdk.md (hunk 1 above)
Comment (gewarren) [inline docs/core/whats-new/dotnet-9/sdk.md:122]:
```suggestion
If you encounter any issues, such as restore failures or unexpected package versions, you can [revert to the legacy resolver](/nuget/consume-packages/Package-References-in-Project-Files#nuget-dependency-resolver).
```

**Review by gewarren — APPROVED** [review]
Thanks - I left a wording suggestion and also a link trim.

**Pull request merged**